### PR TITLE
feat(TDP-4136): add basic validation of preparation before save

### PR DESCRIPTION
**Pull request by hmdebenque**
When a preparation is directly supplied by the frontend, this adds a basic
validation on the preparation object to avoid having null rowMetadata
and then NPE in MetadataChangesOnActionsGenerator.java:73

https://jira.talendforge.org/browse/TDP-4136

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/497/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/497/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/498/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/498/)

File: dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationService.java [hunk 1]
```diff
@@ -442,7 +443,7 @@ public void delete(String id) {
      * @return the updated preparation id.
      */
     public String update(String id, final Preparation preparation) {
```
Comment (fhuaulme) [inline dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationService.java]: Can you use javax.validation.Valid ? And add the javax.validation.constraints.NotNull checks on Preparation?

Comment (hmdebenque) [inline dataprep-preparation/src/main/java/org/talend/dataprep/preparation/service/PreparationService.java]: I was not inclined to use this method as Preparation class is so widely used. I will submit a version using this method.

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/499/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/499/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/502/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/502/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/506/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/506/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/535/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/535/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/556/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/556/)

**Pull request merged**